tl

```diff
@@ -236,7 +236,9 @@ end
 
 local function process_module(filename, env)
    local module_name = filename_to_module_name(filename)
-   local result, err = tl.process(filename, env)
+   local fd
+   if filename == '-' then fd = io.input() end
+   local result, err = tl.process(filename, env, fd)
```

Comment:
This will process standard input with a filename of `-` (for error messages, etc.)

Should we change `filename` to something like `"(standard input)"` here? What do other compilers do?

Reply:
gcc uses `<stdin>`

```shell
$ echo '#include "test.c"' | gcc -x c -
<stdin>:1:10: fatal error: test.c: No such file or directory
compilation terminated.
```

Reply:
Lua itself calls it `stdin` too:

```
echo "if bug then then" | lua -
lua: stdin:1: unexpected symbol near 'then'
```

I like the idea of using some "special symbol" like angle brackets or parentheses to indicate that it's not a real filename (even though technically those _are_ valid filename characters). I have no preference here, up to you!

Reply:
Changed it to use `<stdin>` and `<stdout>` now.